components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImplCopy.java

```diff
@@ -0,0 +1,362 @@
+                errorCode = handleErrorResponseConstants.getResponseMap().get(code)[1];
+                isStatusOnly = false;
+            }
+        }
+        throw buildResourceEndpointException(status, errorCode, throwable == null ? "" : throwable.getMessage(),
+                isStatusOnly);
+    }
+
+    private ResourceEndpointException buildResourceEndpointException(Response.Status status,
+                                                                     String errorCode, String description,
+                                                                     boolean isStatusOnly) {
+        if (isStatusOnly) {
+            return new ResourceEndpointException(status);
+        } else {
+            ErrorDTO errorDTO = new ErrorDTO();
+            errorDTO.setCode(errorCode);
+            errorDTO.setDescription(description);
+            return new ResourceEndpointException(status, errorDTO);
+        }
+    }
+
+    /**
+     * Validate resourceId
+     *
+     * @param resourceId resourceId of the resource which need to get deleted
+     * @return validator obtain
+     */
+    private static final boolean isResourceIdValid(String resourceId) throws UMAClientException {
+
+        String validPattern = "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$";
+        Pattern pattern = Pattern.compile(validPattern);
+        Matcher match = pattern.matcher(resourceId);
+
+        if (match.find()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Retrieve tenantId
+     */
+    private static int getTenantIdFromCarbonContext() {
+
+        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+    }
+
+    /**
+     * Retrieve resource owner name
+     *
+     * @param context message context
+     * @return resource owner name
+     */
+    private String getResourceOwner(MessageContext context) {
+
+        String resourceOwnerName = ((AuthenticationContext) context.getHttpServletRequest().getAttribute(AUTH_CONTEXT))
+                .getUser().getUserName();
+        return resourceOwnerName;
+    }
+
+    /**
+     * Retrieve ConsumerKey
+     *
+     * @param context message context
+     * @return clientId
+     */
+    private String getConsumerKey(MessageContext context) {
+
+        String clientId = (String) ((AuthenticationContext) context.getHttpServletRequest().getAttribute(AUTH_CONTEXT))
+                .getParameter("consumer-key");
+        return clientId;
+    }
+
+    /**
+     * Retrieve Scopes
+     *
+     * @param context message context
+     * @return scopes
+     */
+    private boolean isValidateToken(MessageContext context) {
+
+        String[] tokenScopes = (String[]) ((AuthenticationContext) context.getHttpServletRequest()
+                .getAttribute(AUTH_CONTEXT)).getParameter(OAUTH2_ALLOWED_SCOPES);
+        if (!ArrayUtils.contains(tokenScopes, PAT_SCOPE)) {
```

Comment:
Better to return true if ArrayUtils.contains(tokenScopes, PAT_SCOPE) and false otherwise. It will give a better relationship to the method name.